MonkeyLoader/Configuration/DefiningConfigKey.cs

```diff
@@ -104,13 +102,17 @@ public DefiningConfigKey(string id, string? description = null, Func<T>? compute
 
             AsUntyped = new ConfigKey(id);
 
-            Description = description;
-            HasDescription = !string.IsNullOrWhiteSpace(description);
+            if (description is not null)
+                Components.Add(new ConfigKeyDescription(description));
+
+            if (computeDefault is not null)
+                Components.Add(new ConfigKeyDefault<T>(computeDefault));
+
+            if (valueValidator is not null)
+                Components.Add(new ConfigKeyValidator<T>(valueValidator));
```

Comment:
It should tell you something that I had to read all the way to here to understand wtf is going on with all the super generic entities and components and what have you... Docs with examples would be nice :sweat_smile:

Reply:
Yea, it turned out a bit more complicated due to funny type stuff 😂